Okta.Auth.Sdk/AuthenticationClient.cs

```diff
@@ -57,23 +30,9 @@ public AuthenticationClient(OktaClientConfiguration apiClientConfiguration = nul
         /// </param>
         /// <param name="httpClient">The HTTP client to use for requests to the Okta API.</param>
         /// <param name="logger">The logging interface to use, if any.</param>
-        public AuthenticationClient(OktaClientConfiguration apiClientConfiguration, HttpClient httpClient, ILogger logger = null)
+        public AuthenticationClient(OktaClientConfiguration apiClientConfiguration = null, HttpClient httpClient = null, ILogger logger = null)
+            : base(apiClientConfiguration, httpClient, logger, "okta-auth-dotnet")
         {
-            Configuration = GetConfigurationOrDefault(apiClientConfiguration);
-            OktaClientConfigurationValidator.Validate(Configuration);
-
-            logger = logger ?? NullLogger.Instance;
-
-            var requestExecutor = new DefaultRequestExecutor(Configuration, httpClient, logger);
-            var resourceFactory = new ResourceFactory(this, logger, new AbstractResourceTypeResolverFactory(ResourceTypeHelper.GetAllDefinedTypes(typeof(Resource))));
-            var userAgentBuilder = new UserAgentBuilder("okta-auth-dotnet", typeof(BaseOktaClient).GetTypeInfo().Assembly.GetName().Version);
```

Comment:
The `UserAgentBuilder` receives a version param which is also specific to each SDK. In this case, using the `BaseOktaClient` looks fine (as it lives in the same solution than the Okta.Auth.Sdk) but I believe that it would be better if it takes the `AuthenticationClient` type as versions can be different between the `Okta.Abstractions` and `Okta.Auth.Sdk` packages. That said, the management SDK should create a new user agent builder using `Okta.Sdk.OktaClient` type.

Reply:
I've fixed the way this is passed in so the child implementations pass in their own UserAgentBuilder based on their own version.